analysis/make-epochs.py

```diff
@@ -0,0 +1,125 @@
+    subject_epoch_dir.mkdir(exist_ok=True)
+    # Generate all epoch paths so we can check whether any new TRFs need to be estimated
+    epoch_path = subject_epoch_dir / f'{subject}_epoched_word.pickle'
+    # Skip this subject if all files already exist
+    if epoch_path.exists():
+        continue
+    
+    # Load the EEG data
+    raw = mne.io.read_raw(EEG_DIR / subject / f'{subject}_alice-raw.fif', preload=True)
+    # Band-pass filter the raw data between 0.2 and 20 Hz
+    raw.filter(0.5, 20)
+    # Interpolate bad channels
+    raw.interpolate_bads()
+    # Extract the events marking the stimulus presentation from the EEG file
+    events = eelbrain.load.fiff.events(raw)
+    # Not all subjects have all trials; determine which stimuli are present
+    trial_indexes = [STIMULI.index(stimulus) for stimulus in events['event']]
+    # Extract the EEG data segments corresponding to the stimuli
+    trial_durations = [durations[i] for i in trial_indexes]
+    eeg = eelbrain.load.fiff.variable_length_epochs(events, -0.100, trial_durations, decim=5, connectivity='auto')
+    # Since trials are of unequal length, we will concatenate them for the TRF estimation.
+    eeg_concatenated = eelbrain.concatenate(eeg)
+    
+    # Get corresponding word onsets
+    word_onset_predictor = eelbrain.concatenate([word_onsets[i] for i in trial_indexes])
+    word_surprisal_predictor = eelbrain.concatenate([word_surprisal[i] for i in trial_indexes])
+    
+    rnn_predictor = eelbrain.concatenate([word_RNN[i] for i in trial_indexes])
+    cfg_predictor = eelbrain.concatenate([word_CFG[i] for i in trial_indexes])
+    ngram_preditor = eelbrain.concatenate([word_NGRAM[i] for i in trial_indexes])
+    word_surprisal_predictor = eelbrain.concatenate([word_surprisal[i] for i in trial_indexes])
+    
+    content_word_predictor = eelbrain.concatenate([word_lexical[i] for i in trial_indexes])
+    
+    if word_onset_predictor.time.tstop > eeg_concatenated.time.tstop: 
+        print('error with subject %s' % subject)
+
+    # Make epoched data
+    rows = []
+    for trial_idx, onset_time in enumerate(word_onset_predictor.flatnonzero()):
+        
+        # remark: tstart is negative!
+        if onset_time + tstart < 0: 
+            continue 
+            
+        if onset_time + tstop > eeg_concatenated.time.tstop: 
+            continue
+        
+        current_epoch = eeg_concatenated.sub(time=(onset_time+tstart, onset_time+tstop))
+        # change dimension (tmin to tstart)
+        current_epoch = eelbrain.set_tmin(current_epoch, tmin = tstart)
+        rows.append([current_epoch, word_surprisal_predictor.sub(time=onset_time), trial_idx + 1, 
+                    rnn_predictor.sub(time=onset_time), ngram_preditor.sub(time=onset_time), cfg_predictor.sub(time=onset_time), 
+                    content_word_predictor.sub(time=onset_time)])
+
+    column_names = ['eeg', 'surprisal', 'trial_idx', 'rnn','ngram','cfg', 'content_word']
```

Comment:
Unless you're planning to use them in a different analysis, I would keep it minimal here to avoid confusion and not create and save variables that will never be used.